src/elements/search/search.tsx

```diff
@@ -0,0 +1,22 @@
+import React, { SetStateAction } from "react";
+import styles from "./search.module.scss";
+import { filterProductInfos, selectProductInfos } from "@/api/clientRequests/getProductInfos";
+import { ProductInfo } from "@/productInfos";
+
+interface SearchProps {
+  setResponse: (args: SetStateAction<ProductInfo[] | null>) => Promise<void>;
+}
+
+const Search = (props: SearchProps) => {
+  const onChange = async (event: React.ChangeEvent<HTMLInputElement>) => {
+    if (event.target.value.length !== 0) await filterProductInfos(event.target.value, undefined, props.setResponse);
+    else await selectProductInfos(3, "date", props.setResponse);
+  };
+  return (
+    <form className={styles.search}>
+      <input className={styles.searchLine} placeholder="Search" onChange={onChange} />
+    </form>
+  );
```

Comment:
search works incorrect; 
I want to see game, when i entered part of name; for example when I type "punk" i want to see cyberpunk as result. Use .includes method in filtration for this

Reply:
server filtration function method changed from .startsWith to .includes, as you said